rewrite-core/src/test/java/org/openrewrite/internal/ListUtilsTest.java

```diff
@@ -30,6 +31,13 @@ void flatMap() {
           .containsExactly(1.0, 2.0, 2.1, 2.2, 3.0);
     }
 
+    @Test
+    void flatMapWithNoChangeShouldHaveReferenceEquality() {
+        var before = List.of(1, 2, 3);
+        var after = ListUtils.flatMap(before, Collections::singletonList);
+        assertThat(before == after).isTrue();
```

Comment:
Just a stylistic thing, but if you wanted to be more idiomatic with AssertJ, this could be written as:

```
assertThate(before).isSameAs(after)
```

Which does the same instance equality check.

Reply:
Good point, thanks! 
revised.